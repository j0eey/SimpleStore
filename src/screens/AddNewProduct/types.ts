import { Asset } from "react-native-image-picker";
import { LocationType } from "../../types/types";

export interface FormHeaderProps {
  title: string;
}

export interface ImageUploadSectionProps {
  images: Asset[];
  onImagePick: () => void;
  onCameraCapture: () => void;
  onRemoveImage: (index: number) => void;
  isSubmitting: boolean;
}

export interface FormInputProps {
  label: string;
  value: string;
  onChangeText: (text: string) => void;
  placeholder: string;
  multiline?: boolean;
  numberOfLines?: number;
  keyboardType?: 'default' | 'numeric';
  isSubmitting: boolean;
}

export interface LocationSelectorProps {
  location: LocationType;
  onLocationSelect: () => void;
  isSubmitting: boolean;
}

export interface FormActionsProps {
  onSubmit: () => void;
  onCancel: () => void;
  isSubmitting: boolean;
}