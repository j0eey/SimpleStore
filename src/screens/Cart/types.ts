import { Animated } from "react-native";
import { CartItemType } from "../../types/Cart";

export interface CartHeaderProps {
  itemCount: number;
}

export interface EmptyCartProps {
  onStartShopping: () => void;
}

export interface CartItemProps {
  item: CartItemType;
  onIncreaseQuantity: (itemId: string, currentQuantity: number) => void;
  onDecreaseQuantity: (itemId: string, currentQuantity: number) => void;
  onRemoveItem: (itemId: string) => void;
}

export interface CartSummaryProps {
  cart: CartItemType[];
  summaryExpanded: boolean;
  summaryHeight: Animated.Value;
  onToggleSummary: () => void;
  onCheckout: () => void;
  onClearCart: () => void;
  calculateTotal: () => string;
}

export interface CartListProps {
  cart: CartItemType[];
  onIncreaseQuantity: (itemId: string, currentQuantity: number) => void;
  onDecreaseQuantity: (itemId: string, currentQuantity: number) => void;
  onRemoveItem: (itemId: string) => void;
}