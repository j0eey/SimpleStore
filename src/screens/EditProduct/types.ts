import { Asset } from "react-native-image-picker";
import { LocationType } from "../../types/types";


export interface LoadingScreenProps {}

export interface FormHeaderProps {
  title: string;
}

export interface ImageUploadSectionProps {
  canAddMoreImages: boolean;
  isSubmitting: boolean;
  onGalleryPress: () => void;
  onCameraPress: () => void;
}

export interface ImagePreviewSectionProps {
  existingImages: string[];
  imagesToDelete: string[];
  newImages: Asset[];
  isSubmitting: boolean;
  onRemoveExisting: (imageUrl: string) => void;
  onRemoveNew: (index: number) => void;
}

export interface FormFieldProps {
  label: string;
  value: string;
  onChangeText: (text: string) => void;
  placeholder: string;
  multiline?: boolean;
  numberOfLines?: number;
  keyboardType?: 'default' | 'numeric';
  editable?: boolean;
}

export interface LocationPickerProps {
  location: LocationType;
  isSubmitting: boolean;
  onPress: () => void;
}

export interface FormActionsProps {
  isSubmitting: boolean;
  onSubmit: () => void;
  onCancel: () => void;
  disabled: boolean;
}
