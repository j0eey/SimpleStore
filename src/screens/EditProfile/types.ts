import { EditableProfile } from "../../types/types";

export interface ProfileImageProps {
  imageUrl?: string;
  onPress?: () => void;
}

export interface EditFormProps {
  editedProfile: EditableProfile;
  onProfileChange: (profile: EditableProfile) => void;
  onSave: () => void;
  onCancel: () => void;
  updating: boolean;
}

export interface ImagePickerModalProps {
  visible: boolean;
  onClose: () => void;
  onSelectFromCamera: () => void;
  onSelectFromLibrary: () => void;
}

export interface ProfileSectionProps {
  children: React.ReactNode;
}

export interface MainContainerProps {
  children: React.ReactNode;
}