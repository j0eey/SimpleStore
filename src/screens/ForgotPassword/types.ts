import { Control, FieldErrors } from "react-hook-form";

export interface FormData {
  email: string;
}

export interface FormHeaderProps {
  title: string;
}

export interface EmailInputProps {
  control: Control<FormData>;
  errors: FieldErrors<FormData>;
}

export interface SubmitButtonProps {
  onPress: () => void;
  loading: boolean;
}

export interface LoginLinkProps {
  onPress: () => void;
}

export interface MainContainerProps {
  children: React.ReactNode;
}