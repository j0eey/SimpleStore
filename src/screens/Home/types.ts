import { Product } from "../../types/Product";

export interface ProductItemProps {
  item: Product;
  onPress: (product: Product) => void;
  onAddToCart: (product: Product, position: {x: number, y: number, width: number, height: number}) => void;
  isOwner: boolean;
}

export interface CategoryListProps {
  loading: boolean;
  onCategoryPress: (categoryName: string) => void;
}

export interface LocationSelectorProps {
  selectedLocation: string;
  onSelect: (location: string) => void;
}

export interface ErrorDisplayProps {
  errorMessage: string;
  onRetry: () => void;
}

export interface HomeHeaderProps {
  onProfilePress: () => void;
}