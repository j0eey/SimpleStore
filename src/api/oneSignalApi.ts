import Config from 'react-native-config';
import { OneSignalNotificationPayload, OneSignalApiResponse } from "../types/Product";


class OneSignalApi {
  private restApiKey = Config.ONESIGNAL_REST_API_KEY;
  private baseUrl = 'https://api.onesignal.com';

  /**
   * Send notification via OneSignal REST API
   */
  async sendNotification(payload: OneSignalNotificationPayload): Promise<OneSignalApiResponse> {
    const response = await fetch(`${this.baseUrl}/notifications`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': `Key ${this.restApiKey}`,
      },
      body: JSON.stringify(payload),
    });

    if (!response.ok) {
      throw new Error(`OneSignal API error: ${response.status} ${response.statusText}`);
    }

    return await response.json();
  }
}

export default new OneSignalApi();