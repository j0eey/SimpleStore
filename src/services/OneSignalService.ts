import Config from 'react-native-config';
import oneSignalApi from '../api/oneSignalApi';
import { ProductNotificationData, NotificationResult } from "../types/Product";


class OneSignalService {
  private appId = Config.ONESIGNAL_APP_ID;

  /**
   * Send push notification to all users when a new product is added
   */
  async showProductAddedNotification(product: ProductNotificationData): Promise<NotificationResult> {
    try {
      const notificationPayload = {
        app_id: this.appId,
        included_segments: ['All'], // Send to all subscribed users
        headings: { en: 'New Product Added! ðŸŽ‰' },
        contents: { en: `${product.title} - $${product.price} in ${product.location.name}` },
        data: {
          productId: product._id,
          productTitle: product.title,
          type: 'product_added',
          price: product.price.toString(),
          location: product.location.name,
        },
        buttons: [
          { id: 'view_product', text: 'View Product' },
          { id: 'dismiss', text: 'Dismiss' },
        ],
        big_picture: product.images.length > 0 ? product.images[0].fullUrl : undefined,
        ios_attachments: product.images.length > 0 ? {
          product_image: product.images[0].fullUrl
        } : undefined,
      };

      const result = await oneSignalApi.sendNotification(notificationPayload);

      if (result.id && result.id !== '') {
        return { 
          success: true, 
          notificationId: result.id,
          productId: product._id,
          productTitle: product.title
        };
      } else {
        return { success: false, error: result };
      }
      
    } catch (error) {
      return { 
        success: false, 
        error: error instanceof Error ? error.message : String(error)
      };
    }
  }
}

export default new OneSignalService();