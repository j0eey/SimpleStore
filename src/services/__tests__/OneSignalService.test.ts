import OneSignalService from '../OneSignalService';
import oneSignalApi from '../../api/oneSignalApi';
import { ProductNotificationData } from '../../types/Product';

// Mock react-native-config
jest.mock('react-native-config', () => ({
  ONESIGNAL_APP_ID: 'test-app-id-12345',
}));

// Mock oneSignalApi
jest.mock('../../api/oneSignalApi', () => ({
  sendNotification: jest.fn(),
}));

const mockOneSignalApi = oneSignalApi as jest.Mocked<typeof oneSignalApi>;

describe('OneSignalService', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('showProductAddedNotification', () => {
    const mockProduct: ProductNotificationData = {
      _id: 'product-123',
      title: 'Amazing Product',
      price: 99.99,
      location: {
        name: 'New York, NY',
        latitude: 40.7128,
        longitude: -74.0060,
      },
      images: [
        {
          url: 'image1.jpg',
          _id: 'img-1',
          fullUrl: 'https://example.com/image1.jpg',
        },
        {
          url: 'image2.jpg',
          _id: 'img-2',
          fullUrl: 'https://example.com/image2.jpg',
        },
      ],
    };

    // SNAPSHOT TESTS - for notification payload structures
    describe('Notification Payload Snapshots', () => {
      beforeEach(() => {
        const mockApiResponse = { id: 'test-id-123', recipients: 100 };
        mockOneSignalApi.sendNotification.mockResolvedValue(mockApiResponse);
      });

      it('should match snapshot for standard product notification payload', async () => {
        await OneSignalService.showProductAddedNotification(mockProduct);
        
        const sentPayload = mockOneSignalApi.sendNotification.mock.calls[0][0];
        expect(sentPayload).toMatchSnapshot('standard-product-notification');
      });

      it('should match snapshot for product without images', async () => {
        const productWithoutImages: ProductNotificationData = {
          ...mockProduct,
          images: [],
        };

        await OneSignalService.showProductAddedNotification(productWithoutImages);
        
        const sentPayload = mockOneSignalApi.sendNotification.mock.calls[0][0];
        expect(sentPayload).toMatchSnapshot('product-without-images');
      });

      it('should match snapshot for product with special characters', async () => {
        const specialProduct: ProductNotificationData = {
          ...mockProduct,
          title: 'Test & "Special" Product ðŸŽ‰',
          location: {
            name: 'SÃ£o Paulo, Brazil',
            latitude: -23.5505,
            longitude: -46.6333,
          },
        };

        await OneSignalService.showProductAddedNotification(specialProduct);
        
        const sentPayload = mockOneSignalApi.sendNotification.mock.calls[0][0];
        expect(sentPayload).toMatchSnapshot('product-with-special-characters');
      });

      it('should match snapshot for different price formats', async () => {
        const testProducts = [
          { ...mockProduct, price: 0 },
          { ...mockProduct, price: 10 },
          { ...mockProduct, price: 99.5 },
          { ...mockProduct, price: 1234.56 },
        ];

        const payloads = [];
        for (const product of testProducts) {
          await OneSignalService.showProductAddedNotification(product);
          payloads.push(mockOneSignalApi.sendNotification.mock.calls.pop()?.[0]);
        }

        expect(payloads).toMatchSnapshot('different-price-formats');
      });

      it('should match snapshot for single image product', async () => {
        const singleImageProduct: ProductNotificationData = {
          ...mockProduct,
          images: [{ 
            url: 'single.jpg',
            _id: 'img-single',
            fullUrl: 'https://example.com/single.jpg' 
          }],
        };

        await OneSignalService.showProductAddedNotification(singleImageProduct);
        
        const sentPayload = mockOneSignalApi.sendNotification.mock.calls[0][0];
        expect(sentPayload).toMatchSnapshot('single-image-product');
      });

      it('should match snapshot for edge case scenarios', async () => {
        const edgeCaseProducts = [
          // Empty title
          { ...mockProduct, title: '' },
          // Empty location
          { ...mockProduct, location: { name: '', latitude: 0, longitude: 0 } },
          // Very long title
          { ...mockProduct, title: 'This is a very long product title that exceeds normal length limits and should still be handled correctly by the notification system' },
          // Zero price
          { ...mockProduct, price: 0 },
        ];

        const payloads = [];
        for (const product of edgeCaseProducts) {
          await OneSignalService.showProductAddedNotification(product);
          payloads.push(mockOneSignalApi.sendNotification.mock.calls.pop()?.[0]);
        }

        expect(payloads).toMatchSnapshot('edge-case-scenarios');
      });
    });

    // SNAPSHOT TESTS - for service response structures
    describe('Service Response Snapshots', () => {
      it('should match snapshot for successful response', async () => {
        const mockApiResponse = {
          id: 'notification-id-123',
          recipients: 100,
        };
        mockOneSignalApi.sendNotification.mockResolvedValue(mockApiResponse);

        const result = await OneSignalService.showProductAddedNotification(mockProduct);
        expect(result).toMatchSnapshot('successful-service-response');
      });

      it('should match snapshot for failed responses', async () => {
        const failureScenarios = [
          // Empty ID
          { id: '', error: 'Invalid app ID' },
          // No ID
          { error: 'Authentication failed' },
        ];

        const results = [];
        for (const apiResponse of failureScenarios) {
          mockOneSignalApi.sendNotification.mockResolvedValue(apiResponse);
          const result = await OneSignalService.showProductAddedNotification(mockProduct);
          results.push(result);
        }

        expect(results).toMatchSnapshot('failed-service-responses');
      });

      it('should match snapshot for error handling scenarios', async () => {
        const errorScenarios = [
          new Error('Network request failed'),
          'Something went wrong',
          { code: 500, message: 'Server error' },
        ];

        const results = [];
        for (const error of errorScenarios) {
          mockOneSignalApi.sendNotification.mockRejectedValue(error);
          const result = await OneSignalService.showProductAddedNotification(mockProduct);
          results.push(result);
        }

        expect(results).toMatchSnapshot('error-handling-responses');
      });
    });

    // UNIT TESTS - for behavior and functionality
    describe('Service Behavior Tests', () => {
      describe('Successful Notifications', () => {
        it('should send notification successfully with all product data', async () => {
          const mockApiResponse = {
            id: 'notification-id-123',
            recipients: 100,
          };

          mockOneSignalApi.sendNotification.mockResolvedValue(mockApiResponse);

          const result = await OneSignalService.showProductAddedNotification(mockProduct);

          expect(result).toEqual({
            success: true,
            notificationId: 'notification-id-123',
            productId: 'product-123',
            productTitle: 'Amazing Product',
          });
        });

        it('should call oneSignalApi with correct payload structure', async () => {
          const mockApiResponse = { id: 'test-id' };
          mockOneSignalApi.sendNotification.mockResolvedValue(mockApiResponse);

          await OneSignalService.showProductAddedNotification(mockProduct);

          expect(mockOneSignalApi.sendNotification).toHaveBeenCalledWith({
            app_id: 'test-app-id-12345',
            included_segments: ['All'],
            headings: { en: 'New Product Added! ðŸŽ‰' },
            contents: { en: 'Amazing Product - $99.99 in New York, NY' },
            data: {
              productId: 'product-123',
              productTitle: 'Amazing Product',
              type: 'product_added',
              price: '99.99',
              location: 'New York, NY',
            },
            buttons: [
              { id: 'view_product', text: 'View Product' },
              { id: 'dismiss', text: 'Dismiss' },
            ],
            big_picture: 'https://example.com/image1.jpg',
            ios_attachments: {
              product_image: 'https://example.com/image1.jpg'
            },
          });
        });

        it('should handle product with no images correctly', async () => {
          const productWithoutImages: ProductNotificationData = {
            ...mockProduct,
            images: [],
          };

          const mockApiResponse = { id: 'test-id' };
          mockOneSignalApi.sendNotification.mockResolvedValue(mockApiResponse);

          await OneSignalService.showProductAddedNotification(productWithoutImages);

          expect(mockOneSignalApi.sendNotification).toHaveBeenCalledWith(
            expect.objectContaining({
              big_picture: undefined,
              ios_attachments: undefined,
            })
          );
        });
      });

      describe('Failed Notifications', () => {
        it('should handle API failure with empty ID', async () => {
          const mockApiResponse = {
            id: '',
            error: 'Invalid app ID',
          };

          mockOneSignalApi.sendNotification.mockResolvedValue(mockApiResponse);

          const result = await OneSignalService.showProductAddedNotification(mockProduct);

          expect(result).toEqual({
            success: false,
            error: mockApiResponse,
          });
        });

        it('should handle network errors', async () => {
          const networkError = new Error('Network request failed');
          mockOneSignalApi.sendNotification.mockRejectedValue(networkError);

          const result = await OneSignalService.showProductAddedNotification(mockProduct);

          expect(result).toEqual({
            success: false,
            error: 'Network request failed',
          });
        });

        it('should handle non-Error exceptions', async () => {
          const stringError = 'Something went wrong';
          mockOneSignalApi.sendNotification.mockRejectedValue(stringError);

          const result = await OneSignalService.showProductAddedNotification(mockProduct);

          expect(result).toEqual({
            success: false,
            error: 'Something went wrong',
          });
        });
      });

      describe('Configuration and Payload Validation', () => {
        beforeEach(() => {
          const mockApiResponse = { id: 'test-id' };
          mockOneSignalApi.sendNotification.mockResolvedValue(mockApiResponse);
        });

        it('should include correct app_id from config', async () => {
          await OneSignalService.showProductAddedNotification(mockProduct);

          expect(mockOneSignalApi.sendNotification).toHaveBeenCalledWith(
            expect.objectContaining({
              app_id: 'test-app-id-12345',
            })
          );
        });

        it('should target all users with included_segments', async () => {
          await OneSignalService.showProductAddedNotification(mockProduct);

          expect(mockOneSignalApi.sendNotification).toHaveBeenCalledWith(
            expect.objectContaining({
              included_segments: ['All'],
            })
          );
        });

        it('should convert price to string in data', async () => {
          const productWithNumberPrice: ProductNotificationData = {
            ...mockProduct,
            price: 123.45,
          };

          await OneSignalService.showProductAddedNotification(productWithNumberPrice);

          expect(mockOneSignalApi.sendNotification).toHaveBeenCalledWith(
            expect.objectContaining({
              data: expect.objectContaining({
                price: '123.45', // Should be string
              }),
            })
          );
        });

        it('should include action buttons', async () => {
          await OneSignalService.showProductAddedNotification(mockProduct);

          expect(mockOneSignalApi.sendNotification).toHaveBeenCalledWith(
            expect.objectContaining({
              buttons: [
                { id: 'view_product', text: 'View Product' },
                { id: 'dismiss', text: 'Dismiss' },
              ],
            })
          );
        });
      });

      describe('Image Handling', () => {
        beforeEach(() => {
          const mockApiResponse = { id: 'test-id' };
          mockOneSignalApi.sendNotification.mockResolvedValue(mockApiResponse);
        });

        it('should use first image for big_picture and iOS attachments', async () => {
          await OneSignalService.showProductAddedNotification(mockProduct);

          expect(mockOneSignalApi.sendNotification).toHaveBeenCalledWith(
            expect.objectContaining({
              big_picture: 'https://example.com/image1.jpg',
              ios_attachments: {
                product_image: 'https://example.com/image1.jpg',
              },
            })
          );
        });

        it('should handle products with empty image array', async () => {
          const noImagesProduct: ProductNotificationData = {
            ...mockProduct,
            images: [],
          };

          await OneSignalService.showProductAddedNotification(noImagesProduct);

          expect(mockOneSignalApi.sendNotification).toHaveBeenCalledWith(
            expect.objectContaining({
              big_picture: undefined,
              ios_attachments: undefined,
            })
          );
        });
      });
    });
  });
});