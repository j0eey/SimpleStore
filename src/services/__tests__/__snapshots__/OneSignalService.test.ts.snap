// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OneSignalService showProductAddedNotification Notification Payload Snapshots should match snapshot for different price formats: different-price-formats 1`] = `
[
  {
    "app_id": "test-app-id-12345",
    "big_picture": "https://example.com/image1.jpg",
    "buttons": [
      {
        "id": "view_product",
        "text": "View Product",
      },
      {
        "id": "dismiss",
        "text": "Dismiss",
      },
    ],
    "contents": {
      "en": "Amazing Product - $0 in New York, NY",
    },
    "data": {
      "location": "New York, NY",
      "price": "0",
      "productId": "product-123",
      "productTitle": "Amazing Product",
      "type": "product_added",
    },
    "headings": {
      "en": "New Product Added! ðŸŽ‰",
    },
    "included_segments": [
      "All",
    ],
    "ios_attachments": {
      "product_image": "https://example.com/image1.jpg",
    },
  },
  {
    "app_id": "test-app-id-12345",
    "big_picture": "https://example.com/image1.jpg",
    "buttons": [
      {
        "id": "view_product",
        "text": "View Product",
      },
      {
        "id": "dismiss",
        "text": "Dismiss",
      },
    ],
    "contents": {
      "en": "Amazing Product - $10 in New York, NY",
    },
    "data": {
      "location": "New York, NY",
      "price": "10",
      "productId": "product-123",
      "productTitle": "Amazing Product",
      "type": "product_added",
    },
    "headings": {
      "en": "New Product Added! ðŸŽ‰",
    },
    "included_segments": [
      "All",
    ],
    "ios_attachments": {
      "product_image": "https://example.com/image1.jpg",
    },
  },
  {
    "app_id": "test-app-id-12345",
    "big_picture": "https://example.com/image1.jpg",
    "buttons": [
      {
        "id": "view_product",
        "text": "View Product",
      },
      {
        "id": "dismiss",
        "text": "Dismiss",
      },
    ],
    "contents": {
      "en": "Amazing Product - $99.5 in New York, NY",
    },
    "data": {
      "location": "New York, NY",
      "price": "99.5",
      "productId": "product-123",
      "productTitle": "Amazing Product",
      "type": "product_added",
    },
    "headings": {
      "en": "New Product Added! ðŸŽ‰",
    },
    "included_segments": [
      "All",
    ],
    "ios_attachments": {
      "product_image": "https://example.com/image1.jpg",
    },
  },
  {
    "app_id": "test-app-id-12345",
    "big_picture": "https://example.com/image1.jpg",
    "buttons": [
      {
        "id": "view_product",
        "text": "View Product",
      },
      {
        "id": "dismiss",
        "text": "Dismiss",
      },
    ],
    "contents": {
      "en": "Amazing Product - $1234.56 in New York, NY",
    },
    "data": {
      "location": "New York, NY",
      "price": "1234.56",
      "productId": "product-123",
      "productTitle": "Amazing Product",
      "type": "product_added",
    },
    "headings": {
      "en": "New Product Added! ðŸŽ‰",
    },
    "included_segments": [
      "All",
    ],
    "ios_attachments": {
      "product_image": "https://example.com/image1.jpg",
    },
  },
]
`;

exports[`OneSignalService showProductAddedNotification Notification Payload Snapshots should match snapshot for edge case scenarios: edge-case-scenarios 1`] = `
[
  {
    "app_id": "test-app-id-12345",
    "big_picture": "https://example.com/image1.jpg",
    "buttons": [
      {
        "id": "view_product",
        "text": "View Product",
      },
      {
        "id": "dismiss",
        "text": "Dismiss",
      },
    ],
    "contents": {
      "en": " - $99.99 in New York, NY",
    },
    "data": {
      "location": "New York, NY",
      "price": "99.99",
      "productId": "product-123",
      "productTitle": "",
      "type": "product_added",
    },
    "headings": {
      "en": "New Product Added! ðŸŽ‰",
    },
    "included_segments": [
      "All",
    ],
    "ios_attachments": {
      "product_image": "https://example.com/image1.jpg",
    },
  },
  {
    "app_id": "test-app-id-12345",
    "big_picture": "https://example.com/image1.jpg",
    "buttons": [
      {
        "id": "view_product",
        "text": "View Product",
      },
      {
        "id": "dismiss",
        "text": "Dismiss",
      },
    ],
    "contents": {
      "en": "Amazing Product - $99.99 in ",
    },
    "data": {
      "location": "",
      "price": "99.99",
      "productId": "product-123",
      "productTitle": "Amazing Product",
      "type": "product_added",
    },
    "headings": {
      "en": "New Product Added! ðŸŽ‰",
    },
    "included_segments": [
      "All",
    ],
    "ios_attachments": {
      "product_image": "https://example.com/image1.jpg",
    },
  },
  {
    "app_id": "test-app-id-12345",
    "big_picture": "https://example.com/image1.jpg",
    "buttons": [
      {
        "id": "view_product",
        "text": "View Product",
      },
      {
        "id": "dismiss",
        "text": "Dismiss",
      },
    ],
    "contents": {
      "en": "This is a very long product title that exceeds normal length limits and should still be handled correctly by the notification system - $99.99 in New York, NY",
    },
    "data": {
      "location": "New York, NY",
      "price": "99.99",
      "productId": "product-123",
      "productTitle": "This is a very long product title that exceeds normal length limits and should still be handled correctly by the notification system",
      "type": "product_added",
    },
    "headings": {
      "en": "New Product Added! ðŸŽ‰",
    },
    "included_segments": [
      "All",
    ],
    "ios_attachments": {
      "product_image": "https://example.com/image1.jpg",
    },
  },
  {
    "app_id": "test-app-id-12345",
    "big_picture": "https://example.com/image1.jpg",
    "buttons": [
      {
        "id": "view_product",
        "text": "View Product",
      },
      {
        "id": "dismiss",
        "text": "Dismiss",
      },
    ],
    "contents": {
      "en": "Amazing Product - $0 in New York, NY",
    },
    "data": {
      "location": "New York, NY",
      "price": "0",
      "productId": "product-123",
      "productTitle": "Amazing Product",
      "type": "product_added",
    },
    "headings": {
      "en": "New Product Added! ðŸŽ‰",
    },
    "included_segments": [
      "All",
    ],
    "ios_attachments": {
      "product_image": "https://example.com/image1.jpg",
    },
  },
]
`;

exports[`OneSignalService showProductAddedNotification Notification Payload Snapshots should match snapshot for product with special characters: product-with-special-characters 1`] = `
{
  "app_id": "test-app-id-12345",
  "big_picture": "https://example.com/image1.jpg",
  "buttons": [
    {
      "id": "view_product",
      "text": "View Product",
    },
    {
      "id": "dismiss",
      "text": "Dismiss",
    },
  ],
  "contents": {
    "en": "Test & "Special" Product ðŸŽ‰ - $99.99 in SÃ£o Paulo, Brazil",
  },
  "data": {
    "location": "SÃ£o Paulo, Brazil",
    "price": "99.99",
    "productId": "product-123",
    "productTitle": "Test & "Special" Product ðŸŽ‰",
    "type": "product_added",
  },
  "headings": {
    "en": "New Product Added! ðŸŽ‰",
  },
  "included_segments": [
    "All",
  ],
  "ios_attachments": {
    "product_image": "https://example.com/image1.jpg",
  },
}
`;

exports[`OneSignalService showProductAddedNotification Notification Payload Snapshots should match snapshot for product without images: product-without-images 1`] = `
{
  "app_id": "test-app-id-12345",
  "big_picture": undefined,
  "buttons": [
    {
      "id": "view_product",
      "text": "View Product",
    },
    {
      "id": "dismiss",
      "text": "Dismiss",
    },
  ],
  "contents": {
    "en": "Amazing Product - $99.99 in New York, NY",
  },
  "data": {
    "location": "New York, NY",
    "price": "99.99",
    "productId": "product-123",
    "productTitle": "Amazing Product",
    "type": "product_added",
  },
  "headings": {
    "en": "New Product Added! ðŸŽ‰",
  },
  "included_segments": [
    "All",
  ],
  "ios_attachments": undefined,
}
`;

exports[`OneSignalService showProductAddedNotification Notification Payload Snapshots should match snapshot for single image product: single-image-product 1`] = `
{
  "app_id": "test-app-id-12345",
  "big_picture": "https://example.com/single.jpg",
  "buttons": [
    {
      "id": "view_product",
      "text": "View Product",
    },
    {
      "id": "dismiss",
      "text": "Dismiss",
    },
  ],
  "contents": {
    "en": "Amazing Product - $99.99 in New York, NY",
  },
  "data": {
    "location": "New York, NY",
    "price": "99.99",
    "productId": "product-123",
    "productTitle": "Amazing Product",
    "type": "product_added",
  },
  "headings": {
    "en": "New Product Added! ðŸŽ‰",
  },
  "included_segments": [
    "All",
  ],
  "ios_attachments": {
    "product_image": "https://example.com/single.jpg",
  },
}
`;

exports[`OneSignalService showProductAddedNotification Notification Payload Snapshots should match snapshot for standard product notification payload: standard-product-notification 1`] = `
{
  "app_id": "test-app-id-12345",
  "big_picture": "https://example.com/image1.jpg",
  "buttons": [
    {
      "id": "view_product",
      "text": "View Product",
    },
    {
      "id": "dismiss",
      "text": "Dismiss",
    },
  ],
  "contents": {
    "en": "Amazing Product - $99.99 in New York, NY",
  },
  "data": {
    "location": "New York, NY",
    "price": "99.99",
    "productId": "product-123",
    "productTitle": "Amazing Product",
    "type": "product_added",
  },
  "headings": {
    "en": "New Product Added! ðŸŽ‰",
  },
  "included_segments": [
    "All",
  ],
  "ios_attachments": {
    "product_image": "https://example.com/image1.jpg",
  },
}
`;

exports[`OneSignalService showProductAddedNotification Service Response Snapshots should match snapshot for error handling scenarios: error-handling-responses 1`] = `
[
  {
    "error": "Network request failed",
    "success": false,
  },
  {
    "error": "Something went wrong",
    "success": false,
  },
  {
    "error": "[object Object]",
    "success": false,
  },
]
`;

exports[`OneSignalService showProductAddedNotification Service Response Snapshots should match snapshot for failed responses: failed-service-responses 1`] = `
[
  {
    "error": {
      "error": "Invalid app ID",
      "id": "",
    },
    "success": false,
  },
  {
    "error": {
      "error": "Authentication failed",
    },
    "success": false,
  },
]
`;

exports[`OneSignalService showProductAddedNotification Service Response Snapshots should match snapshot for successful response: successful-service-response 1`] = `
{
  "notificationId": "notification-id-123",
  "productId": "product-123",
  "productTitle": "Amazing Product",
  "success": true,
}
`;
